test/gtest/common/test_helpers.h

```diff
@@ -384,17 +384,35 @@ class message_stream {
     ~message_stream();
 
     template <typename T>
-    std::ostream& operator<<(const T& value) const {
-        return std::cout << value;
+    message_stream& operator<<(const T& value) {
+        msg << value;
+        return *this;
+    }
+
+    message_stream& operator<< (std::ostream&(*f)(std::ostream&)) {
+        if (f == (std::basic_ostream<char>& (*)(std::basic_ostream<char>&)) &std::flush) {
```

Comment:
As we discussed, this is not a C++ way

Reply:
but it works ok with std::flush, so seems good enough. What is the problem?

Reply:
It's not a problem from functionality point of view but it's hard to read like C++ code and adds extra check at runtime :)